File: cryptopals/set1/challenge5.py
def repeating_key_xor(b: bytes, key: bytes) -> bytearray:
    result = bytearray()
    for i in range(len(b)):
        result.append(b[i] ^ key[i % len(key)])
    return result


def main():
    s = "Burning 'em, if you ain't quick and nimble\nI go crazy when I hear a cymbal"
    print(repeating_key_xor(bytes(s, "ascii"), bytes("ICE", "ascii")).hex())


if __name__ == "__main__":
    main()

File: set1/challenge5.go
package set1

import (
	"encoding/hex"
	"fmt"
)

func Challenge5() {
	str := "Burning 'em, if you ain't quick and nimble\nI go crazy when I hear a cymbal"
	key := "ICE"

	encoded := make([]byte, len(str))
	for i := 0; i < len(str); i += 1 {
		encoded[i] = str[i] ^ key[(i%len(key))]
	}

	fmt.Println(hex.EncodeToString(encoded))
}
